### PR TITLE
Feature/marketplace minor enhancements

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

two things: 

- install/update directly the plugin after a cloud download
- differenciante local update from new installation

image

### CODE REVIEW

File: inc/plugin.class.php [hunk 1]
```diff
@@ -922,6 +922,31 @@ function isActivated($directory) {
...
 
 
+   /**
+    * Is a plugin updatable ?
+    *
+    * @param string $directory  Plugin directory
+    *
+    * @return boolean
+    */
+   function isUpdatable($directory) {
```
Comment (trasher) [inline inc/plugin.class.php:932]: Maybe this new method could be tested?

File: inc/plugin.class.php [hunk 2]
```diff
@@ -922,6 +922,31 @@ function isActivated($directory) {
...
+      // Make a lowercase comparison, as sometime this function is called based on
+      // extraction of plugin name from a classname, which does not use same naming rules than directories.
+      $activated_plugins = array_map('strtolower', self::$activated_plugins);
+      if (in_array(strtolower($directory), $activated_plugins)) {
+         // If plugin is marked as activated, no need to query DB on this case.
+         return true;
```
Comment (cedric-anne) [inline inc/plugin.class.php:938]: If plugin is active, it should mean that its state cannot be `NOTUPDATED`.
```suggestion
         return false;
```

File: inc/marketplace/view.class.php [hunk 3]
```diff
@@ -658,7 +659,7 @@ static function getButtons(string $plugin_key = ""): string {
          } else if ($can_be_updated) {
```
Comment (cedric-anne) [inline inc/marketplace/view.class.php:659]: Maybe button title should be changed by `Download and install` 2 lines above.

Comment (orthagh) [reply inc/marketplace/view.class.php:659]: i try, as much as possible, to create new locales

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**